onboarding_project/squad_pantry_app/models.py

```diff
@@ -0,0 +1,48 @@
+from django.db import models
+from datetime import datetime
+from django.contrib.auth.models import User
+from django.core.validators import MinValueValidator
+
+# Create your models here.
+
+
+class Dish(models.Model):
+    dish_name = models.CharField(max_length=256, unique=True)
+    DISH_TYPE = (
+        ('NV', 'Non-Vegetarian'),
+        ('V', 'Vegetarian'),
+        ('E', 'Contains Egg'),
+    )
+    dish_type = models.CharField(max_length=2, choices=DISH_TYPE, default=DISH_TYPE[0][0])
+    is_available = models.BooleanField(default=False, blank=False)
+    estimated_preparation_time_in_minutes = models.IntegerField(validators=[MinValueValidator(1)])
+
+
+class Order(models.Model):
+    user = models.ForeignKey(User, on_delete=models.CASCADE)
+    STATUS = (
+        ('OP', 'Order Placed'),
+        ('REJ', 'Rejected'),
+        ('ACC', 'Accepted'),
+        ('CAN', 'Cancelled'),
+        ('PROC', 'Processing'),
+        ('DEL', 'Delivered')
+    )
+    status = models.CharField(max_length=4, choices=STATUS, default=STATUS[0][0])
+    scheduled_time = models.DateTimeField(blank=True, null=True)
+    order_placed_at = models.DateTimeField(default=datetime.now())
+    order_closed_at = models.DateTimeField(blank=True, null=True)
+
+
+class OrderDishRelation(models.Model):
+    order = models.ForeignKey(Order, on_delete=models.CASCADE)
+    dish = models.ForeignKey(Dish, on_delete=models.CASCADE)
+    quantity = models.IntegerField(validators=[MinValueValidator(1)])
+
+    class Meta:
+        unique_together = ["order", "dish"]
```

Comment:
Will this work? Official [django doc](https://docs.djangoproject.com/en/2.0/ref/models/options/#unique-together) uses tuple. I did not test this, see if the unique constraint is actually honored.

Reply:
Yup... Working absolutely fine.
![screen shot 2018-02-02 at 5 13 51 pm](https://user-images.githubusercontent.com/13600114/35731661-90ce6f30-083c-11e8-925e-1992e49aa4ea.png)